packages/td-tools/src/td-parser.ts

```diff
@@ -237,32 +240,20 @@ export function serializeTD(thing: Thing): string {
         delete copy.properties;
     } else if (copy.properties != null) {
         // add mandatory fields (if missing): observable, writeOnly, and readOnly
-        for (const propName in copy.properties) {
-            const prop = copy.properties[propName];
-            if (prop.readOnly === undefined || typeof prop.readOnly !== "boolean") {
-                prop.readOnly = false;
-            }
-            if (prop.writeOnly === undefined || typeof prop.writeOnly !== "boolean") {
-                prop.writeOnly = false;
-            }
-            if (prop.observable === undefined || typeof prop.observable !== "boolean") {
-                prop.observable = false;
-            }
+        for (const prop of Object.values(copy.properties)) {
+            prop.readOnly ??= false;
+            prop.writeOnly ??= false;
+            prop.observable ??= false;
```

Comment:
I am not 100% sure but I think the *new* code is not exactly doing the same. The old code fixed wrong types while the new one does not do that. Or am I wrong?

e.g., `"readOnly": "YES"` .. was changed to` "readOnly": false`

Question is also whether we want that to happen..

Reply:
Thank you, that should be covered via a9d20792a0cebfb3f43aea0cc73730c7800a5dec now :)